### PR TITLE
autoflex: Add object map and mapof

### PR DESCRIPTION
**Pull request by YakDriver**
### Relations

Closes #0000

### Output from Acceptance Testing

```console
% go test ./internal/framework/flex/... -v -count 1 -parallel 20 -run='TestGenericExpand2/map'
=== RUN   TestGenericExpand2
=== PAUSE TestGenericExpand2
=== CONT  TestGenericExpand2
=== RUN   TestGenericExpand2/map_string
=== PAUSE TestGenericExpand2/map_string
=== RUN   TestGenericExpand2/object_map
=== PAUSE TestGenericExpand2/object_map
=== RUN   TestGenericExpand2/object_map_ptr_target
=== PAUSE TestGenericExpand2/object_map_ptr_target
=== RUN   TestGenericExpand2/object_map_ptr_source_and_target
=== PAUSE TestGenericExpand2/object_map_ptr_source_and_target
=== RUN   TestGenericExpand2/nested_object_map
=== PAUSE TestGenericExpand2/nested_object_map
=== CONT  TestGenericExpand2/map_string
=== CONT  TestGenericExpand2/object_map_ptr_source_and_target
=== CONT  TestGenericExpand2/object_map_ptr_target
=== CONT  TestGenericExpand2/object_map
=== CONT  TestGenericExpand2/nested_object_map
--- PASS: TestGenericExpand2 (0.00s)
    --- PASS: TestGenericExpand2/object_map (0.00s)
    --- PASS: TestGenericExpand2/object_map_ptr_target (0.00s)
    --- PASS: TestGenericExpand2/object_map_ptr_source_and_target (0.00s)
    --- PASS: TestGenericExpand2/map_string (0.00s)
    --- PASS: TestGenericExpand2/nested_object_map (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/framework/flex	0.361s
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Stack trace (from running in the debugger) when the the error `Diagnostic` is raised:

```
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromAttributeValue (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/interfaces.go:323)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/outof.go:26)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromStruct (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/struct.go:217)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/outof.go:57)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromPointer (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/pointer.go:118)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/outof.go:85)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromSlice (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/slice.go:162)
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/reflect/outof.go:71)
github.com/hashicorp/terraform-plugin-framework/types/basetypes.NewListValueFrom (/Users/ewbankkit/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/types/basetypes/list_value.go:100)
github.com/hashicorp/terraform-provider-aws/internal/framework/types.newListNestedObjectValueOf[go.shape.struct { Field1 github.com/hashicorp/terraform-plugin-framework/types/basetypes.MapValue "tfsdk:\"field1\"" }] (/Users/ewbankkit/altsrc.2/github.com/terraform-providers/terraform-provider-aws/internal/framework/types/list_nested_objectof.go:257)
github.com/hashicorp/terraform-provider-aws/internal/framework/types.NewListNestedObjectValueOfSlice[go.shape.struct { Field1 github.com/hashicorp/terraform-plugin-framework/types/basetypes.MapValue "tfsdk:\"field1\"" }] (/Users/ewbankkit/altsrc.2/github.com/terraform-providers/terraform-provider-aws/internal/framework/types/list_nested_objectof.go:249)
github.com/hashicorp/terraform-provider-aws/internal/framework/types.NewListNestedObjectValueOfPtr[go.shape.struct { Field1 github.com/hashicorp/terraform-plugin-framework/types/basetypes.MapValue "tfsdk:\"field1\"" }] (/Users/ewbankkit/altsrc.2/github.com/terraform-providers/terraform-provider-aws/internal/framework/types/list_nested_objectof.go:245)
github.com/hashicorp/terraform-provider-aws/internal/framework/flex.TestGenericExpand (/Users/ewbankkit/altsrc.2/github.com/terraform-providers/terraform-provider-aws/internal/framework/flex/autoflex_test.go:544)
testing.tRunner (/Users/ewbankkit/sdk/go1.20.10/src/testing/testing.go:1576)
testing.(*T).Run.func1 (/Users/ewbankkit/sdk/go1.20.10/src/testing/testing.go:1629)
runtime.goexit (/Users/ewbankkit/sdk/go1.20.10/src/runtime/asm_amd64.s:1598)
```

File: internal/framework/flex/autoflex.go [hunk 1]
```diff
@@ -550,7 +556,23 @@ func (expander autoExpander) mapOfString(ctx context.Context, vFrom basetypes.Ma
...
+					diags.Append(vFrom.ElementsAs(ctx, &vTo, true)...)
+					if diags.HasError() {
+						return diags
+					}
+
+					vTo.Set(reflect.ValueOf(vTo))
+					return diags
 				}
+			default:
+				diags.Append(vFrom.ElementsAs(ctx, &tMapElem, false)...)
```
Comment (ewbankkit) [inline internal/framework/flex/autoflex.go:569]: Is there a useful comment that could be added here?

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% go test ./internal/framework/types
ok  	github.com/hashicorp/terraform-provider-aws/internal/framework/types	1.114s
% go test ./internal/framework/flex 
ok  	github.com/hashicorp/terraform-provider-aws/internal/framework/flex	1.804s
```

File: internal/framework/flex/autoflex.go [hunk 2]
```diff
@@ -550,7 +556,23 @@ func (expander autoExpander) mapOfString(ctx context.Context, vFrom basetypes.Ma
 
 					vTo.Set(reflect.ValueOf(to))
 					return diags
+				default:
+					diags.Append(vFrom.ElementsAs(ctx, &vTo, true)...)
```
[the remark this answers is not in the document]
Comment (YakDriver) [reply internal/framework/flex/autoflex.go:560]: That was part of debugging and shouldn't be there. Thanks for catching!

File: internal/framework/flex/autoflex.go (hunk 1 above)
Comment (YakDriver) [reply internal/framework/flex/autoflex.go:569]: That was part of debugging and shouldn't be there. Thanks for catching!

**Pull request merged**